program/src/processor.rs

```diff
@@ -442,14 +442,31 @@ fn process_harvest_rewards(program_id: &Pubkey, accounts: &[AccountInfo]) -> Pro
     // Since the holder rewards account may already have unharvested rewards,
     // calculate the share of rewards that have not been seen by the holder
     // rewards account.
-    let unseen_total_rewards = current_total_rewards
-        .checked_sub(last_seen_total_rewards)
-        .ok_or(ProgramError::ArithmeticOverflow)?;
-    let unseen_unharvested_rewards =
-        calculate_reward_share(token_supply, token_account_balance, unseen_total_rewards)?;
-    let rewards_to_harvest = unharvested_rewards
-        .checked_add(unseen_unharvested_rewards)
-        .ok_or(ProgramError::ArithmeticOverflow)?;
+    let rewards_to_harvest = {
+        // Rewards accrued by the system that haven't been seen by the holder.
+        let unseen_total_rewards = current_total_rewards.saturating_sub(last_seen_total_rewards);
```

Comment:
Is there a reason to use saturating here? checked would be safer since total rewards should only increase

Reply:
Yeah we can use checked here. I feel like this warrants its own error, though, rather than `ArithmeticOverflow`. What do you think?

Reply:
I don't feel strongly either way, so feel free to add a new error if you think it's better